subsys/net/ip/net_if.c

```diff
@@ -555,6 +561,90 @@ static void dad_timeout(struct k_work *work)
 		 */
 		net_ipv6_nbr_rm(iface, &ifaddr->address.in6_addr);
 	}
+
+	ifaddr->dad.timer_timeout = 0;
+
+	sys_slist_find_and_remove(&active_dad_timers, &ifaddr->dad.node);
+}
+
+static bool dad_check_timeout(s64_t start, u32_t time, s64_t timeout)
+{
+	start += time;
+	start = abs(start);
+
+	if (start > timeout) {
+		return false;
+	}
+
+	return true;
+}
+
+static bool dad_timedout(struct net_if_addr *ifaddr, s64_t timeout)
+{
+	return dad_check_timeout(ifaddr->dad.timer_start,
+				 ifaddr->dad.timer_timeout,
+				 timeout);
+}
+
+static u32_t dad_manage_timeouts(struct net_if_addr *ifaddr, s64_t timeout)
```

Comment:
Why this is called manage_timeout**s**, while it's apparently deals with a single one?

Reply:
Then in should be named like "manage_and_return_timeouts". But as long as it manages a *single* timeout, the naming is still confusing ;-).